code/modules/projectiles/guns/projectile/shotgun/bull.dm

```diff
@@ -48,22 +61,22 @@
 	if(chambered)
 		chambered.loc = get_turf(src)//Eject casing
 		chambered = null
-		if(reload == 0)
-			if(loaded.len)
-				var/obj/item/ammo_casing/AC = loaded[1] //load next casing.
-				loaded -= AC //Remove casing from loaded list.
-				chambered = AC
-	reload = 1
+		if(other_chambered != null)
+			chambered = other_chambered //load the main barrel we use with the side barrel.
+			other_chambered = null
 
-/obj/item/weapon/gun/projectile/shotgun/bull/unload_ammo(user, allow_dump)
+/obj/item/weapon/gun/projectile/shotgun/bull/unload_ammo(mob/user, allow_dump)
 	if(chambered)
+		to_chat(user, SPAN_NOTICE("You unchamber [(other_chambered ? 2 : 1)] round\s from [src]."))
```

Comment:
```suggestion
		to_chat(user, SPAN_NOTICE("You unchamber [(other_chambered ? 2 : 1)] shell\s from [src]."))
```

Reply:
this code for a second barrel seems more than a little snowflake-y, i'll have a thorough look through, but i'm not convinced this is the best way to do it.

Perhaps a better approach would be to implement a barrels var for all guns